tasks/hyrax-metadata-updates/tests/test-index.js

```diff
@@ -0,0 +1,123 @@
+});
+
+test('Test return prod OPeNDAP host when ops environment value supplied', async (t) => {
+  process.env.CMR_ENVIRONMENT = 'OPS';
+  const actual = await generateAddress();
+  t.is(actual, 'https://opendap.earthdata.nasa.gov');
+});
+
+test('Test return sit OPeNDAP host when sit environment value supplied', async (t) => {
+  process.env.CMR_ENVIRONMENT = 'SIT';
+  const actual = await generateAddress();
+  t.is(actual, 'https://opendap.sit.earthdata.nasa.gov');
+});
+
+test('Test return uat OPeNDAP host when uat environment value supplied', async (t) => {
+  process.env.CMR_ENVIRONMENT = 'UAT';
+  const actual = await generateAddress();
+  t.is(actual, 'https://opendap.uat.earthdata.nasa.gov');
+});
+
+test('Test return error when invalid environment supplied for host generation', async (t) => {
+  process.env.CMR_ENVIRONMENT = 'FOO';
+  t.throws(
+    () => generateAddress(),
+    {
+      message: 'Environment foo is not a valid environment.',
+      instanceOf: InvalidArgument
+    }
+  );
+});
+
+test('Test granule ur extraction from UMM-G', async (t) => {
+  const data = fs.readFileSync('tests/data/umm-gin.json', 'utf8');
+  const metadata = JSON.parse(data);
+  const actual = await getGranuleUr(metadata, true);
+
+  t.is(actual, 'GLDAS_CLSM025_D.2.0:GLDAS_CLSM025_D.A20141230.020.nc4');
+});
+
+test('Test granule ur extraction from ECHO10', async (t) => {
+  const data = fs.readFileSync('tests/data/echo10in.xml', 'utf8');
+  const metadata = await (promisify(xml2js.parseString))(data, xmlParseOptions);
+  const actual = await getGranuleUr(metadata, false);
+
+  t.is(actual, 'GLDAS_CLSM025_D.2.0:GLDAS_CLSM025_D.A20141230.020.nc4');
+});
+
+test('Test adding OPeNDAP URL to UMM-G file', async (t) => {
+  const data = fs.readFileSync('tests/data/umm-gin.json', 'utf8');
+  const metadata = JSON.parse(data);
+  const expected = fs.readFileSync('tests/data/umm-gout.json', 'utf8');
+  const expectedObject = JSON.parse(expected);
+  const actual = await addHyraxUrl(metadata, true, 'https://opendap.earthdata.nasa.gov/providers/GES_DISC/collections/GLDAS%20Catchment%20Land%20Surface%20Model%20L4%20daily%200.25%20x%200.25%20degree%20V2.0%20(GLDAS_CLSM025_D)%20at%20GES%20DISC/granules/GLDAS_CLSM025_D.2.0:GLDAS_CLSM025_D.A20141230.020.nc4');
+  t.is(actual, JSON.stringify(expectedObject, null, 2));
+});
+
+test('Test adding OPeNDAP URL to UMM-G file with no related urls', async (t) => {
+  const data = fs.readFileSync('tests/data/umm-gin-no-related-urls.json', 'utf8');
+  const metadata = JSON.parse(data);
+  const expected = fs.readFileSync('tests/data/umm-gout-no-related-urls.json', 'utf8');
+  const expectedObject = JSON.parse(expected);
+  const actual = await addHyraxUrl(metadata, true, 'https://opendap.earthdata.nasa.gov/providers/GES_DISC/collections/GLDAS%20Catchment%20Land%20Surface%20Model%20L4%20daily%200.25%20x%200.25%20degree%20V2.0%20(GLDAS_CLSM025_D)%20at%20GES%20DISC/granules/GLDAS_CLSM025_D.2.0:GLDAS_CLSM025_D.A20141230.020.nc4');
```

Comment:
I don't think this `await` is necessary, nor that the test function needs to be `async`.